onnxruntime/core/framework/execution_context.h

```diff
@@ -0,0 +1,173 @@
+#pragma once
+#include "core/common/logging/logging.h"
+#include "core/framework/device_stream_collection.h"
+#include "core/framework/execution_frame.h"
+#include "core/framework/ort_value.h"
+#include "core/framework/iexecutor.h"
+#include "core/framework/stream_handles.h"
+#include "core/graph/basic_types.h"
+#include "core/common/inlined_containers.h"
+#include "core/framework/memory_info.h"
+#ifdef ENABLE_TRAINING
+#include "core/framework/partial_graph_execution_state.h"
+#endif
+
+namespace onnxruntime {
+class SessionState;
+
+class SessionScope;
+typedef InlinedHashMap<std::string, OrtValue> OrtValueCache;
+typedef std::shared_ptr<OrtValueCache> OrtValueCachePtr;
+
+// Execution context that support to execute a command on stream.
+// It is composed by following components:
+// 1. a execution frame
+// 2. a collection of device stream instances that kernels can launch to.
+// 3. a set of notification instances needed to perform synchronization in current execution plan.
+class ExecutionContext {
```

Comment:
Would prefer calling it StreamExecutionContext since ExecutionContext is too general and can be confusing. If you rename this, please update the filename as well.